compiler/qsc_parse/src/lex/cooked.rs

```diff
@@ -51,6 +51,10 @@ pub(crate) enum Error {
     #[error("unrecognized character `{0}`")]
     #[diagnostic(code("Qsc.Lex.UnknownChar"))]
     Unknown(char, #[label] Span),
+
+    #[error("unfinished generic")]
+    #[diagnostic(code("Qsc.Lex.UnfinishedGeneric"))]
+    UnfinishedGeneric(#[label] Span),
```

Comment:
This should follow the same pattern as the Incomplete and IncompleteEof errors above.